tests/func/update/security_group.yml

```diff
@@ -0,0 +1,53 @@
+- name: change item in file for update test
+  lineinfile:
+    dest: "{{ os_migrate_data_dir }}/security_groups.yml"
+    regexp: "description: OSM security group"
+    line: "    description: 'OSM security group updated'"
+    state: present
+    backup: yes
+    backrefs: yes
+
+- name: scan available security_groups after before test
+  os_migrate.os_migrate.os_security_groups_info:
+    filters: "{{ os_migrate_dst_filters }}"
+    auth: "{{ os_migrate_dst_auth }}"
+    auth_type: "{{ os_migrate_dst_auth_type|default(omit) }}"
+    region_name: "{{ os_migrate_dst_region_name|default(omit) }}"
+    validate_certs: "{{ os_migrate_dst_validate_certs|default(omit) }}"
+    ca_cert: "{{ os_migrate_dst_ca_cert|default(omit) }}"
+    client_cert: "{{ os_migrate_dst_client_cert|default(omit) }}"
+    client_key: "{{ os_migrate_dst_client_key|default(omit) }}"
+  register: dst_sg_info_before
+
+- include_role:
+    name: os_migrate.os_migrate.import_security_groups
+
+- name: scan available security_groups after update test
+  os_migrate.os_migrate.os_security_groups_info:
+    filters: "{{ os_migrate_dst_filters }}"
+    auth: "{{ os_migrate_dst_auth }}"
+    auth_type: "{{ os_migrate_dst_auth_type|default(omit) }}"
+    region_name: "{{ os_migrate_dst_region_name|default(omit) }}"
+    validate_certs: "{{ os_migrate_dst_validate_certs|default(omit) }}"
+    ca_cert: "{{ os_migrate_dst_ca_cert|default(omit) }}"
+    client_cert: "{{ os_migrate_dst_client_cert|default(omit) }}"
+    client_key: "{{ os_migrate_dst_client_key|default(omit) }}"
+  register: dst_sg_info_after
+
+- name: original description for osm_security_group
+  debug:
+    msg: "{{ (dst_sg_info_before['openstack_security_groups']
+      | json_query(\"[?name=='osm_security_group'].description\"))[0] }}"
+
+- name: updated description for osm_security_group
+  debug:
+    msg: "{{ (dst_sg_info_after['openstack_security_groups']
+      | json_query(\"[?name=='osm_security_group'].description\"))[0] }}"
+
+- name: ensure description for osm_security_group changed
+  assert:
+    that:
+      - dst_sg_info_before['openstack_security_groups'] | json_query(\"[?name=='osm_security_group'].description\")[0] == 'OSM security group'
```

Comment:
Maybe we can instead of:
```
json_query(\"[?name=='osm_security_group'].description\")[0]
```
this:
```
json_query("[?name=='osm_security_group'].description")[0]
```

The above rang a bell with me, i've hit a similar issue with escaping earlier. The issue is that `debug: msg` parameter expects a string so it needs "{{ }}" use a jinja expression. But `assert: that` already expects an expression, so we don't wrap it in double quotes so we don't need to escape any double quotes used inside.